### PR TITLE
feat: support bedrock llm provider

### PR DESCRIPTION
**Pull request by Mini256**
As title.

image

### CODE REVIEW

File: backend/app/rag/types.py [hunk 1]
```diff
@@ -23,6 +24,11 @@ class VertexModel(str, enum.Enum):
     CLAUDE_35_SONNET = "claude-3-5-sonnet@20240620"
 
 
+# Reference: https://us-west-2.console.aws.amazon.com/bedrock/home?region=us-west-2#/models
+class BedrockModel(str, enum.Enum):
+    CLAUDE_35_SONNET = "anthropic.claude-3-5-sonnet-20240620-v1:0"
```
Comment (wd0517) [inline backend/app/rag/types.py:29]: These class appears to be deprecated. I will delete it after merging this PR.

File: backend/app/rag/chat_config.py [hunk 2]
```diff
@@ -170,6 +172,26 @@ def get_llm(
...
+                context_size = 200000
+
+            llm = Bedrock(
+                model=model,
+                aws_access_key_id=access_key_id,
+                aws_secret_access_key=secret_access_key,
+                region_name=region_name,
+                context_size=context_size
+            )
+            llm.aws_access_key_id=access_key_id
```
[the remark this answers is not in the document]
Comment (Mini256) [reply backend/app/rag/chat_config.py:191]: In order to pass the values to `get_dspy_lm_by_llama_llm` function, but unfortunately, llamaindex Bedrock class doesn't set up this values to the corresponding attributes in its constructor function.

File: backend/app/api/admin_routes/llm.py [hunk 3]
```diff
@@ -81,6 +82,7 @@ def test_llm(
         success = True
         error = ""
     except Exception as e:
+        logging.exception(e)
```
[the remark this answers is not in the document]
Comment (Mini256) [reply backend/app/api/admin_routes/llm.py:85]: 😢 But it's difficult to locate the place where the error was reported without the stack information.

File: backend/app/alembic/versions/04d81be446c3_.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+               existing_type=mysql.ENUM('OPENAI', 'GEMINI', 'ANTHROPIC_VERTEX', 'OPENAI_LIKE', 'BEDROCK'),
+               nullable=False)
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column('llms', 'provider',
+               existing_type=mysql.ENUM('OPENAI', 'GEMINI', 'ANTHROPIC_VERTEX', 'OPENAI_LIKE'),
+               nullable=True)
```
[the remark this answers is not in the document]
Comment (Mini256) [reply backend/app/alembic/versions/04d81be446c3_.py]: Fixed

File: backend/app/api/admin_routes/llm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (wd0517) [reply backend/app/api/admin_routes/llm.py:85]: ok, how about `logger.debug`, by the way, do not use global logging, use `logger = logging.getLogger(__name__); logger.debug()` instead.

File: backend/app/rag/chat_config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wd0517) [reply backend/app/rag/chat_config.py:191]: add some comment to explain this

**Pull request merged**